### PR TITLE
Use operation system agnostic path separators

### PR DESCRIPTION
**Pull request by fearnlj01**
Fixes issue with windows based devices being unable to pull file based metadata

### CODE REVIEW

File: Shokofin/API/Models/File.cs [hunk 1]
```diff
@@ -80,7 +80,7 @@ public class Location
         /// where the <see cref="File"/> lies, with a leading slash applied at
         /// the start.
         /// </summary>
-        public string Path => "/" + RelativePath;
+        public string Path => System.IO.Path.DirectorySeparatorChar + RelativePath;
```
Comment (revam) [inline Shokofin/API/Models/File.cs:83]: We also need to make sure the path returned from  Shoko is OS agnostic/independent. It is currently dependent on the environment where Shoko Server runs (which may be Windows, Linux, or something else, etc.). Also, since we're doing path transformations we should probably cache the value for later re-use.

```suggestion
        public string Path =>
            __path != null ? (
                __path
             ) : (
                __path = System.IO.Path.DirectorySeparatorChar + RelativePath
                    .Replace('/', System.IO.Path.DirectorySeparatorChar)
                    .Replace('\\', System.IO.Path.DirectorySeparatorChar)
             );

        /// <summary>
        /// Cached path for later re-use.
        /// </summary>
        private string? __path { get; set; }
```

**Review by revam — CHANGES_REQUESTED** [review]
So currently we assume everything runs under an unix-like environment where everything uses the forward slash (/) in paths. The current state of this PR assumes both Jellyfin and Shoko at least runs in the same OS environment (i.e. both runs under Windows or both runs under Linux, etc.), but we also need to assume that Jellyfin and Shoko can run in different environments. I've added a suggestion to also convert the path separators in the path received from Shoko, which is the missing piece of code before we achieve OS agnostic paths for the paths from both the Jellyfin and the Shoko.

Also good job finding this issue on your own and for making a PR about it.

TL;DR: I made a suggestion i want you to approve/commit (in the GitHub UI) before I merge this.

**Review by revam — APPROVED** [review]
:+1:
